doc/developers/advanced_installation.rst

```diff
@@ -229,10 +229,10 @@ console:
 For 64-bit Python, configure the build environment by running the following
 commands in ``cmd`` or an Anaconda Prompt (if you use Anaconda):
 
-    ::
+.. prompt:: bash $
```

Comment:
I was reading https://sphinx-copybutton.readthedocs.io/en/latest/use.html to understand what `copybutton_exclude` does and it mentioned that the right prompt type is `console`, not `bash`. Should we change this here (and in other places) or is `bash` "the right thing" to use here?

Reply:
I may misunderstand, but I think you confuse sphinx-prompt with sphinx-copybutton.

`copybutton_exclude` is a directive to exclude CSS selectors for text extraction from the code block. Since sphinx-prompt automatically inlines a `style` element in the first code block of a page, we want to exclude `style` element otherwise the copied text will not be what we expect.

The only doc about sphinx-prompt I found is: https://pypi.org/project/sphinx-prompt/
I don't see a reference to a `console` "language" (see positional arguments).

`bash` was the thing used in most of the existing scikit-learn documentation. The vanilla alternative seems to be `text`.

Reply:
Sorry, it did not notice the context. For windows you can use either `batch` which gives a `C:\> ` prompt, or `powershell` (`PS C:\> ` prompt).

Here, `batch` would make sense, but then you would have to search for all windows oriented shell commands so the documentation is consistent. I thus recommend to simply leave `bash` for now.

Reply:
I think for this specific command, it only works with `batch`: in particular it would not work in a [Git Bash](https://gitforwindows.org/) terminal, so I think we can make an exception here, and specifically use the `batch` prompt even if we do not update the other windows prompt.

Reply:
OK 👍

Reply:
I had to work around a limitation in a older version of sphinx-prompt. Please let me know if [my comment](https://github.com/scikit-learn/scikit-learn/pull/27136/files#diff-123d1441ce4fd5ffa635f8e1bd96746ff741a5ad17fbd160ea3da354956cb832R232) is appropriate.

Also, I would be interested to know why older versions of sphinx or its plugins need to be supported. Isn't document generation a task private to the scikit-learn project?

Reply:
We want to not force every contributor to have the latest version of every tool because it can lead to unresolvable overconstrained dependencies in the dev environments.

For such dev tools we are not conservative as for the main runtime dependencies of scikit-learn and tend to bump them whenever supporting the old versions becomes too cumbersome from a maintenance point of view. Pragmatic handling of tradeoffs.

Reply:
BTW I think your workaround is good for now.